tests/testthat/test-toolGetMapping.R

```diff
@@ -38,4 +38,12 @@ test_that("toolGetMapping works", {
   writeLines("abc", paste0(getConfig("mappingfolder"), "/test.xyz"))
   expect_error(toolGetMapping("test.rda", where = "mappingfolder"), "did not contain a object")
   expect_error(toolGetMapping("test.xyz"), "Unsupported filetype")
+
+  readTest <- function() {
+    toolGetMapping("test.csv", type = "regional")
+    return(as.magpie(1))
+  }
+  globalassign("readTest")
+  expect_warning(readSource("Test", convert = "onlycorrect"),
+                "No correct function for Test could be found. Set convert to FALSE.")
```

Comment:
This test is not testing the warning you added. I think passing `convert = FALSE` or not passing `convert` at all might already be enough to trigger your warning.

Reply:
I tried both options, but the tests failed.

Reply:
In addition to Pascal's comment, this test is not checking the right warning.